### PR TITLE
SSR Bilateral Filtering

### PR DESCRIPTION
**Pull request by dmcdiarmid-ly**
Added Bilateral Filtering to the SSR roughness blur.
Added Downsampled Linear Depth passes.

Tested Editor DX12/Vulkan
Tested AtomSampleViewer DX12/Vulkan

### CODE REVIEW

File: Gems/Atom/Feature/Common/Assets/Shaders/Reflections/ReflectionScreenSpaceDownsampleDepthLinear.azsl [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+{    
+    float2 m_result : SV_Target0;
+};
+
+PSOutput MainPS(VSOutput IN)
+{
+    uint2 coords = IN.m_position.xy;
+    uint2 previousMipBaseCoords = IN.m_position.xy * 2;
+
+    float minDepth = 1e27f;
```
Comment (moudgils) [inline Gems/Atom/Feature/Common/Assets/Shaders/Reflections/ReflectionScreenSpaceDownsampleDepthLinear.azsl:44]: This function needs comments related to it's purpose and any other specific details

File: Gems/Atom/Feature/Common/Assets/Shaders/Reflections/ReflectionScreenSpaceDownsampleDepthLinear.azsl [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    float sigma = 0.0f;
+    float2 depthSamples[9];
```
Comment (moudgils) [inline Gems/Atom/Feature/Common/Assets/Shaders/Reflections/ReflectionScreenSpaceDownsampleDepthLinear.azsl:46]: Add a comment about why the length is 9

File: Gems/Atom/Feature/Common/Code/Source/ReflectionScreenSpace/ReflectionScreenSpaceDownsampleDepthLinearPass.cpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            ParentPass::BuildInternal();
+
+            // setup attachment bindings on the child passes
+            uint32_t attachmentIndex = 0;
+            for (auto& childPass : m_children)
+            {
+                // the first child pass reads from DepthLinear and writes to mip0 of DownsampledDepthLinear, subsequent
+                // passes read from the previous mip level of DownsampledDepthLinear and write to the current mip level
+                uint32_t currentMipLevel = attachmentIndex;
+                uint32_t previousMipLevel = AZStd::max(0, (int)currentMipLevel - 1);
```
Comment (moudgils) [inline Gems/Atom/Feature/Common/Code/Source/ReflectionScreenSpace/ReflectionScreenSpaceDownsampleDepthLinearPass.cpp:87]: static_cast<int>

**Review by moudgils — APPROVED** [review]

**Review by AMZN-alexpete — APPROVED** [review]

File: Gems/Atom/Feature/Common/Assets/Shaders/Reflections/ReflectionScreenSpaceDownsampleDepthLinear.azsl [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                    sigma = depthSamples[i].g;
+                }
+            }                
+        }
+    }
+
+    PSOutput OUT;
+    OUT.m_result = float2(minDepth, sigma);
+    return OUT;
+}
```
Comment (AMZN-alexpete) [inline Gems/Atom/Feature/Common/Assets/Shaders/Reflections/ReflectionScreenSpaceDownsampleDepthLinear.azsl:148]: nit: missing newline at end of file

File: Gems/Atom/Feature/Common/Code/Source/ReflectionScreenSpace/ReflectionScreenSpaceDownsampleDepthLinearPass.cpp [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        {
+            RemoveChildren();
+        }
+
+        void ReflectionScreenSpaceDownsampleDepthLinearPass::CreateChildPassesInternal()
+        {
+            RPI::PassSystemInterface* passSystem = RPI::PassSystemInterface::Get();
+
+            // load shader
+            AZStd::string shaderFilePath = "Shaders/Reflections/ReflectionScreenSpaceDownsampleDepthLinear.azshader";
```
Comment (AMZN-alexpete) [inline Gems/Atom/Feature/Common/Code/Source/ReflectionScreenSpace/ReflectionScreenSpaceDownsampleDepthLinearPass.cpp:38]: hmm do we need to add the shader to some engine seed list somewhere for release builds?

**Pull request merged**